test/core/storage/trie/trie_db/trie_batch_test.cpp

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/**
+ * @given an empty trie
+ * @when putting some data to it
+ * @then inserted data is accessible
+ */
+
+#include "storage/trie/polkadot_trie_db/polkadot_trie_batch.hpp"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <testutil/storage/map_storage.hpp>
+#include "crypto/hasher/hasher_impl.hpp"
+#include "testutil/literals.hpp"
+#include "testutil/outcome.hpp"
+#include "testutil/storage/base_leveldb_test.hpp"
+#include "testutil/storage/polkadot_trie_db_printer.hpp"
+
+using namespace kagome::storage::trie;
+using kagome::common::Buffer;
+using kagome::hash::HasherImpl;
+using testing::_;
+using testing::Invoke;
+using testing::Return;
+
+class TrieBatchTest : public test::BaseLevelDB_Test {
+ public:
+  TrieBatchTest() : BaseLevelDB_Test("/tmp/leveldbtest") {}
+
+  void SetUp() override {
+    open();
+    trie = std::make_unique<PolkadotTrieDb>(std::move(db_));
+  }
+
+  std::vector<std::pair<Buffer, Buffer>> data = {
+      {"123456"_hex2buf, "42"_hex2buf},
+      {"1234"_hex2buf, "1234"_hex2buf},
+      {"010203"_hex2buf, "0a0b"_hex2buf},
+      {"010a0b"_hex2buf, "1337"_hex2buf},
+      {"0a0b0c"_hex2buf, "deadbeef"_hex2buf}};
+
+  std::unique_ptr<PolkadotTrieDb> trie;
+};
+
+class MockPolkadotTrieDb : public PolkadotTrieDb {
+ public:
+  MockPolkadotTrieDb() : PolkadotTrieDb(nullptr) {}
+
+  MOCK_CONST_METHOD0(getRootHash, Buffer());
+};
+
+class MockDb : public test::MapDb {
+ public:
+  MOCK_METHOD2(put, outcome::result<void>(const Buffer &, const Buffer &));
+
+  // to retain the ability to call the actual implementation of put from the
+  // superclass
+  outcome::result<void> true_put(const Buffer &key, const Buffer &value) {
+    return MapDb::put(key, value);
+  }
+};
```

Comment:
Please, move the mocks to the separate files, one for each mock, as they can be needed in future. They should be in the same places, as the original classes, repeating the structure of the projects inside `/test/mock/` folder

Reply:
These two mocks are very specific, with only one method mocked for each, as it needed in this test suit. I think that if a need for bigger mocks arises, it won't be hard to move these to separate files and expand them. But for now I don't see a need.